File: WordPress/src/androidTest/java/org/wordpress/android/e2e/ContactUsTests.java
package org.wordpress.android.e2e;

import org.junit.Before;
import org.junit.Test;
import org.wordpress.android.e2e.flows.LoginFlow;
import org.wordpress.android.e2e.pages.ContactSupportScreen;
import org.wordpress.android.support.BaseTest;

import static org.wordpress.android.BuildConfig.E2E_WP_COM_USER_EMAIL;

import dagger.hilt.android.testing.HiltAndroidTest;

@HiltAndroidTest
public class ContactUsTests extends BaseTest {
    @Before
    public void setUp() {
        logoutIfNecessary();
    }

    @Test
    public void e2eSendButtonEnabledWhenTextIsEntered() {
        try {
            new LoginFlow()
                .chooseContinueWithWpCom(super.mComposeTestRule)
                .tapHelp()
                .assertHelpScreenLoaded()
                .openContactUs()
                .assertContactSupportScreenLoaded()
                .assertSendButtonDisabled()
                .setMessageText("Hello")
                .assertSendButtonEnabled()
                .setMessageText("")
                .assertSendButtonDisabled();
        } finally {
            new ContactSupportScreen().goBackAndDeleteUnsentMessageIfNeeded();
        }
    }

    @Test
    public void e2eHelpCanBeOpenedWhileEnteringEmail() {
        new LoginFlow()
                .chooseContinueWithWpCom(super.mComposeTestRule)
                .tapHelp()
                .assertHelpScreenLoaded();
    }

    @Test
    public void e2eHelpCanBeOpenedWhileEnteringPassword() {
        new LoginFlow()
                .chooseContinueWithWpCom(super.mComposeTestRule)
                .enterEmailAddress(E2E_WP_COM_USER_EMAIL)
                .tapHelp()
                .assertHelpScreenLoaded();
    }
}

File: WordPress/src/androidTest/java/org/wordpress/android/e2e/ContactUsTests.kt
package org.wordpress.android.e2e

import dagger.hilt.android.testing.HiltAndroidTest
import org.junit.Before
import org.junit.Test
import org.wordpress.android.BuildConfig
import org.wordpress.android.e2e.flows.LoginFlow
import org.wordpress.android.e2e.pages.ContactSupportScreen
import org.wordpress.android.e2e.pages.HelpScreen
import org.wordpress.android.support.BaseTest

@HiltAndroidTest
class ContactUsTests : BaseTest() {
    @Before
    fun setUp() {
        logoutIfNecessary()
    }

    @Test
    fun e2eSendButtonEnabledWhenTextIsEntered() {
        try {
            LoginFlow()
                .chooseContinueWithWpCom(super.mComposeTestRule)
                .tapHelp()
                .assertHelpScreenLoaded()

            // Contact Us is only available on Jetpack App, see: https://github.com/wordpress-mobile/WordPress-Android/pull/18818
            if (BuildConfig.IS_JETPACK_APP) {
                HelpScreen().openContactUs()
                    .assertContactSupportScreenLoaded()
                    .assertSendButtonDisabled()
                    .setMessageText("Hello")
                    .assertSendButtonEnabled()
                    .setMessageText("")
                    .assertSendButtonDisabled()
            }
        } finally {
            ContactSupportScreen().goBackAndDeleteUnsentMessageIfNeeded()
        }
    }

    @Test
    fun e2eHelpCanBeOpenedWhileEnteringEmail() {
        if (BuildConfig.IS_JETPACK_APP) {
            LoginFlow()
                .chooseContinueWithWpCom(super.mComposeTestRule)
                .tapHelp()
                .assertHelpScreenLoaded()
        }
    }

    @Test
    fun e2eHelpCanBeOpenedWhileEnteringPassword() {
        if (BuildConfig.IS_JETPACK_APP) {
            LoginFlow()
                .chooseContinueWithWpCom(super.mComposeTestRule)
                .enterEmailAddress(BuildConfig.E2E_WP_COM_USER_EMAIL)
                .tapHelp()
                .assertHelpScreenLoaded()
        }
    }
}
